### DISCUSSION TITLE
변수에 타입이 고정되어 있나요?

### DISCUSSION THREAD

**Discussion by publdaze** [publdaze/js-study#29] [데이터 타입]
- 변수에 타입이 고정되어 있나요?

**Comment by pipisebastian** [comment]
>변수에 타입이 고정되어 있나요?

자바스크립트에서 변수는 여러 타입이 들어갈 수 있습니다. 자바스크립트는 동적타입언어로, 런타임중에 변경될 수 있습니다.
이는 실제로 기존 값이 담긴 메모리를 변경하는게 아니라, 새로운 메모리를 만들고 새로운 타입의 원시값을 만드는 것입니다.

**Comment by publdaze** [comment]
> - 변수에 타입이 고정되어 있나요?
- 변수에는 타입이 고정되어 있지 않고, 데이터 값에 따라 타입이 정해짐
- 런타임에 값이 할당될 때 데이터 타입이 변경될 수 있으므로 자바스크립트는 동적 타입 언어임
